Tasks/Common/npm-common/npmregistry.ts

```diff
@@ -36,6 +40,12 @@ export class NpmRegistry implements INpmRegistry {
 
         try {
             url = NormalizeRegistry(tl.getEndpointUrl(endpointId, false));
+            if (endpointAuth.scheme === 'Token' &&
+                URL.parse(url).hostname.toUpperCase().endsWith('.VISUALSTUDIO.COM')){
+                // VSTS does not support a PAT in _authToken.  Therefore, you must do basic.
+                endpointAuth.scheme = 'CrossAccount';
```

Comment:
minor, but would be better to treat endpointAuth as immutable and not set a scheme type that it doesn't really support.

Reply:
@pspill, @aldoms , code has been adjusted.